### PR TITLE
Add curl speed limit for low bandwidth server

### PR DESCRIPTION
**Pull request by Kxnrl**

### CODE REVIEW

File: http_natives.cpp [hunk 1]
```diff
@@ -445,6 +509,10 @@ const sp_nativeinfo_t http_natives[] =
 	{"HTTPClient.FollowLocation.set",	SetClientFollowLocation},
 	{"HTTPClient.Timeout.get",			GetClientTimeout},
 	{"HTTPClient.Timeout.set",			SetClientTimeout},
+	{"HTTPClient.MaxSendSpeed.get",		GetClientMaxSendSpeed},
+	{"HTTPClient.MaxSendSpeed.set",		SetClientMaxSendSpeed},
+	{"HTTPClient.MaxRecvSpeed.get",		GetClientMaxRecvSpeed},
+	{"HTTPClient.MaxRecvSpeed.set",		SetClientMaxRecvSpeed},
```
Comment (ErikMinekus) [inline http_natives.cpp:515]: You need to add these properties to the `HTTPClient` methodmap in `http.inc` as well.
Preferably with documentation indicating that the value is in bytes/sec.

File: httpfilecontext.cpp [hunk 2]
```diff
@@ -83,6 +84,15 @@ void HTTPFileContext::InitCurl()
...
 	curl_easy_setopt(curl, CURLOPT_TIMEOUT, timeout);
 	curl_easy_setopt(curl, CURLOPT_URL, url.chars());
+
+	if (maxRecvSpeed > 0)
+	{
+		curl_easy_setopt(curl, CURLOPT_MAX_RECV_SPEED_LARGE, maxRecvSpeed);
+	}
+	if (maxSendSpeed > 0)
+	{
+		curl_easy_setopt(curl, CURLOPT_MAX_RECV_SPEED_LARGE, maxSendSpeed);
```
Comment (ErikMinekus) [inline httpfilecontext.cpp:94]: `CURLOPT_MAX_SEND_SPEED_LARGE`

File: httpfilecontext.h [hunk 3]
```diff
@@ -49,6 +50,8 @@ class HTTPFileContext : public IHTTPContext
 	long connectTimeout;
 	long followLocation;
 	long timeout;
+	long maxSendSpeed;
+	long maxRecvSpeed;
```
Comment (ErikMinekus) [inline httpfilecontext.h:54]: According to the cURL documentation these should be of type `curl_off_t`.

Comment (Kxnrl) [reply httpfilecontext.h:54]: yeap, i will change it later, curl_off_t is int64_t I think.

File: httpfilecontext.cpp (hunk 2 above)
Comment (Kxnrl) [reply httpfilecontext.cpp:94]: haha, my bad.

File: pawn/scripting/include/ripext/http.inc [hunk 4]
```diff
@@ -174,4 +174,16 @@ methodmap HTTPClient < Handle
 		public native get();
 		public native set(int timeout);
 	}
+
+	// Max Send speed in bytes per seconed.
```
Comment (peace-maker) [inline pawn/scripting/include/ripext/http.inc:178]: second*

Same typo below.

Comment (Kxnrl) [reply pawn/scripting/include/ripext/http.inc:178]: ::cry::

**Pull request merged**